src/briefcase/platforms/web/static.py

```diff
@@ -96,33 +116,354 @@ def _trim_file(self, path, sentinel):
+        self,
+        wheel: ZipFile,
+        path: Path,
+        filename: str,
+        package_key: str,
+        inserts: dict[str, dict[str, dict[str, str]]],
+        legacy_css_warning: bool,
+    ) -> bool:
+        """Handle legacy CSS under /static/*.css and add to briefcase.css.
+
+        :param wheel: Open wheel ZipFile being processed.
+        :param path: Path object of the file inside the wheel.
+        :param filename: Filename string inside the wheel.
+        :param package_key: Provenance label (e.g. "name version").
+        :param inserts: Nested dict of inserts keyed by target - insert - package.
+        :param legacy_css_warning: Whether the warning has already been shown.
+        :return: Updated legacy_css_warning flag.
+        """
+        self.console.info(f"    Found {filename}")
+
+        # Show deprecation warning once per wheel
+        if not legacy_css_warning:
+            self.console.warning(
+                f"    {Path(wheel.filename).name}: legacy '/static' CSS detected, "
+                "treating as insert into briefcase.css, this legacy handling will be removed in the future."
+            )
+            legacy_css_warning = True
+
+        css_text = wheel.read(filename).decode("utf-8")
+
+        rel_inside = "/".join(path.parts[2:])
+        contrib_key = f"{package_key} (legacy static CSS: {rel_inside})"
+
+        # Add CSS content to briefcase.css insert slot
+        target = "static/css/briefcase.css"
+        insert = "css"
+        pkg_map = inserts.setdefault(target, {}).setdefault(insert, {})
+        if contrib_key in pkg_map and pkg_map[contrib_key]:
+            pkg_map[contrib_key] += "\n" + css_text
+        else:
+            pkg_map[contrib_key] = css_text
 
-        :param wheelfile: The path to the wheel file to be processed.
-        :param css_file: A file handle, opened for write/append, to which any extracted
-            CSS content will be appended.
+        return legacy_css_warning
+
+    def _handle_insert(
+        self,
+        wheel: ZipFile,
+        parts: tuple[str, ...],
+        filename: str,
+        package_key: str,
+        inserts: dict[str, dict[str, dict[str, str]]],
+    ) -> None:
+        """Handle deploy/inserts/<target>~<insert> files and register inserts.
+
+        :param wheel: Open wheel ZipFile being processed.
+        :param parts: Path parts of the filename inside the wheel.
+        :param filename: Filename string inside the wheel.
+        :param package_key: Provenance label (e.g. "name version").
+        :param inserts: Nested dict of inserts keyed by target - insert - package.
+        """
+        self.console.info(f"    Found {filename}")
+
+        rel_inside = "/".join(parts[3:])
+
+        if not rel_inside or rel_inside.endswith("/"):
+            self.console.debug(f"    {filename}: skipping, not a valid insert file.")
+            return
+
+        if "~" not in rel_inside:
+            self.console.debug(
```

Comment:
These are both enough of a problem that I'd say they're worth flagging as a warning; debug will be essentially hidden unless you go digging, and they're an indication of a badly formed insert.